### ISSUE TITLE
[Paper Note] Small Batch Size Training for Language Models: When Vanilla SGD Works,  and Why Gradient Accumulation Is Wasteful, Martin Marek+, arXiv'25

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#2192]
# URL
- https://arxiv.org/abs/2507.07101
# Authors
  - Martin Marek
  - Sanae Lotfi
  - Aditya Somasundaram
  - Andrew Gordon Wilson
  - Micah Goldblum
# Abstract
  - Conventional wisdom dictates that small batch sizes make language model pretraining and fine-tuning unstable, motivating gradient accumulation, which trades off the number of optimizer steps for a proportional increase in batch size. While it is common to decrease the learning rate for smaller batch sizes, other hyperparameters are often held fixed. In this work, we revisit small batch sizes all the way down to batch size one, and we propose a rule for scaling Adam hyperparameters to small batch sizes. We find that small batch sizes (1) train stably, (2) are consistently more robust to hyperparameter choices, (3) achieve equal or better per-FLOP performance than larger batch sizes, and (4) notably enable stable language model training with vanilla SGD, even without momentum, despite storing no optimizer state. Building on these results, we provide practical recommendations for selecting a batch size and setting optimizer hyperparameters. We further recommend against gradient accumulation unless training on multiple devices with multiple model replicas, bottlenecked by inter-device bandwidth.
# Translation (by gpt-4o-mini)
- 従来の知見では、小さなバッチサイズは言語モデルの事前学習とファインチューニングを不安定にするため、勾配蓄積が推奨されており、これはオプティマイザのステップ数をバッチサイズの比例的な増加と引き換えにする手法である。小さなバッチサイズに対して学習率を下げることは一般的であるが、他のハイパーパラメータはしばしば固定されたままとなる。本研究では、バッチサイズを1まで小さくすることを再考し、小さなバッチサイズに対するAdamのハイパーパラメータをスケーリングするルールを提案する。私たちは、小さなバッチサイズが(1) 安定してトレーニングされ、(2) ハイパーパラメータの選択に対して一貫してよりロバストであり、(3) 大きなバッチサイズと同等またはそれ以上のパフォーマンスを達成し、(4) 特にバニラSGDを用いた安定した言語モデルのトレーニングを可能にすることを発見した。これは、モーメンタムなしでオプティマイザの状態を保存せずに実現される。これらの結果に基づき、バッチサイズの選択とオプティマイザのハイパーパラメータ設定に関する実用的な推奨事項を提供する。また、複数のデバイスで複数のモデルレプリカを使用してトレーニングする場合を除き、勾配蓄積は推奨しない。
# Summary (by gpt-4o-mini)
- 小さなバッチサイズに対するAdamのハイパーパラメータをスケーリングする新しいルールを提案。これにより、小さなバッチサイズでも安定したトレーニングが可能で、大きなバッチサイズと同等以上のパフォーマンスを達成。勾配蓄積は推奨せず、実用的なハイパーパラメータ設定のガイドラインを提供。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/giffmana/status/1943384733418950815?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

論文中のFigure1において、AdamWにおいてbatchsizeが1の方が512の場合と比べてlearning_rateの変化に対してロバストである旨が記述されている。

<img width="977" height="642" alt="Image" src="https://github.com/user-attachments/assets/0c1efb5d-6eeb-4fd7-ba06-e4296e988a6c" />

**Comment by AkihikoWatanabe** [comment]
似たような話でMTでバッチサイズ小さいほうが性能良いです、みたいな話が昔あったような

（追記）
気になって思い出そうとしていたが、MTではなく画像認識の話だったかもしれない（だいぶうろ覚え）
- #2196

**Comment by AkihikoWatanabe** [comment]
参考:https://x.com/odashi_t/status/1944034128707342815?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
関連:
- #1541

**Comment by AkihikoWatanabe** [comment]
解説:https://x.com/hillbig/status/1952506470878351492?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
実際に8Bモデルの事前学習においてβ2を0.99にしたところ、学習が不安定になり、かつ最終的なPerplexityも他の設定に勝つことができなかったとのこと:
https://x.com/odashi_t/status/1955906705637957995?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q
